tests/unit/bootstrap_record_test.cc

```diff
@@ -0,0 +1,269 @@
+using namespace seastar;
+using namespace seastar::fs;
+
+namespace {
+
+constexpr off_t device_size = 1 * MB;
+
+class mock_block_device_impl : public block_device_impl {
+public:
+    using buf_type = basic_sstring<uint8_t, size_t, 32, false>;
+    buf_type buf;
+    ~mock_block_device_impl() override = default;
+
+    future<size_t> write(uint64_t pos, const void* buffer, size_t len, const io_priority_class& pc) override {
+        if (buf.size() < pos + len)
+            buf.resize(pos + len);
+        std::memcpy(buf.data() + pos, buffer, len);
+        return make_ready_future<size_t>(len);
+    }
+
+    future<size_t> read(uint64_t pos, void* buffer, size_t len, const io_priority_class& pc) override {
+        std::memcpy(buffer, buf.data() + pos, len);
+        return make_ready_future<size_t>(len);
+    }
+
+    future<> flush() override {
+        return make_ready_future<>();
+    }
+
+    future<> close() override {
+        return make_ready_future<>();
+    }
+};
+
+future<block_device> prepare_device() {
+    return async([&] {
+        sstring device_path = [&]() -> sstring {
+            temporary_file tmp_device_file("/tmp/bootstrap_record_test_file");
+            return tmp_device_file.path();
+        }();
+
+        file f = open_file_dma(device_path, open_flags::rw | open_flags::create).get0();
+        f.truncate(device_size).get();
+        f.close().get();
+
+        return open_block_device(device_path).get0();
+    });
+}
+
+}
+
+BOOST_TEST_DONT_PRINT_LOG_VALUE(bootstrap_record)
+SEASTAR_TEST_CASE(test_valid_read_write) {
+    return prepare_device().then([] (block_device dev) {
+        return do_with(std::move(dev), [] (block_device& dev) {
+            return async([&] {
+                bootstrap_record write_record(1, 1024, 1024, 1, 3, {{6, {6, 9}}, {9, {9, 12}}, {12, {12, 15}}});
+                // basic test
+                write_record.write_to_disk(dev).get();
+                auto read_record = bootstrap_record::read_from_disk(dev).get0();
+                BOOST_REQUIRE_EQUAL(write_record, read_record);
+
+                // max number of shards
+                write_record.shards_nb = bootstrap_record::max_shards_nb;
+                write_record.shards_info.resize(write_record.shards_nb, {1, {2, 3}});
+                write_record.write_to_disk(dev).get();
+                read_record = bootstrap_record::read_from_disk(dev).get0();
+                BOOST_REQUIRE_EQUAL(write_record, read_record);
+
+                // 0 shards
+                write_record.shards_nb = 0;
+                write_record.shards_info.clear();
+                write_record.write_to_disk(dev).get();
+                read_record = bootstrap_record::read_from_disk(dev).get0();
+                BOOST_REQUIRE_EQUAL(write_record, read_record);
+            }).finally([&] {
+                return dev.close();
+            });
+        });
+    });
+}
+
+namespace {
+
+void repair_crc32(shared_ptr<mock_block_device_impl> dev_impl) {
```

Comment:
```suggestion
inline void repair_crc32(shared_ptr<mock_block_device_impl> dev_impl) noexcept {
```

Reply:
Are you sure about that noexcept?

Reply:
Boost seems to be written in an unreadable way without noexcept specifier anywhere in crc.hpp, so I cannot check it, but I see no reason for it to throw. For me it is unnecessary to allocate memory there or do anything else that would fail, but don't cite me on this.